frameworks/laravel.md

```diff
@@ -0,0 +1,107 @@
+- TypeScript [types for the install event](https://github.com/sfreytag/laravel-vite-pwa/blob/main/resources/js/composables/usePwa/types.ts)
+
+## Setup
+
+The repo has been built on a vanilla install of Laravel 10 using composer from `composer create-project laravel/laravel`.
+
+To add the PWA to your own Laravel project you can review the changes required to set up `vite-plugin-pwa`:
+
+- Work through the commit history, which builds it up step-by-step
+- Or view the entire diff of the HEAD against the vanilla Laravel install: https://github.com/sfreytag/laravel-vite-pwa/compare/a59497..HEAD
+
+Or just fork the repo and start from there.
+
+## Build
+
+To build the repo, follow the usual Laravel steps. Nothing extra is required for `vite-plugin-pwa`. Assuming you have PHP, NPM and Composer:
+
+```
+git clone git@github.com:sfreytag/laravel-vite-pwa.git
+cd laravel-vite-pwa
+composer install
+cp .env.example .env
+php artisan key:generate
+npm install
+npm run build
+```
+
+## Run 
+
+Before you run it, bear in mind that the PWA installs a service worker and fills a cache. This can conflict with other service workers and caches from your other localhost projects. So it is recommended to use a port unique to each PWA project. To use eg 8082 for Laravel:
+
+```
+php artisan serve --port=8082
+```
+
+The app should now be running on `http://localhost:8082`. It should immediately work as a PWA. If you check the dev tools, the service worker should be running. If your browser supports it there will be an intall prompt in the address bar. It should then be installable. And if you use dev tools to take either the network or service worker offline, it should continue working if you reload the page.
+
+## Working on the PWA
+
+The PWA is configured to only work with prod builds, rather than dev. This is straightforward to work with and helps stop the PWA offline cache get in the way of refreshing your build during the dev cycle. However this might not suit everyone. It would be a good PR to submit to this repo to get the PWA working with a dev build. In the meantime, before running the PWA and when making changes, to be sure you have the latest version of it, ensure you use:
+
+```
+npm run build
+```
+
+## PWA Icons
+
+The repo uses [@vite-pwa/assets-generator](https://github.com/vite-pwa/assets-generator) for its icons.
+
+The canonical icon should be an SVG file. This is useful for the PWA anyway, so it can be saved in `public/favicon.svg`. Then build the other icons from it by running:
+
+```
+npm run pwa-icons
+```
+
+This generates a set of icons defined by the minimal preset described [here](/assets-generator/cli.html#presets).
+
+They are automatically packaged in the public folder so they are web readable. They are also included in the repo so this process only needs repeating if you change the canonical `favicon.svg` icon.
+
+## Screenshot
```

Comment:
remove this section, or replace it with a link to your readme.md anchor with the image:

https://github.com/sfreytag/laravel-vite-pwa?tab=readme-ov-file#screenshot